TaskWithClass/test.cpp

```diff
@@ -0,0 +1,270 @@
+#include "pch.h"
+#include "MyString.h"
+
+TEST(StringTest, TestAt)
+{
+	std::string what;
+	String s;
+
+	try
+	{
+		what = "";
+		s.at(0);
+
+	}
+	catch (std::out_of_range const& exc)
+	{
+		what = exc.what();
+	}
+	EXPECT_EQ(std::string("invalid string position"), what);
+
+	s = "abcd";
+	EXPECT_EQ('a', s.at(0));
+	EXPECT_EQ('b', s.at(1));
+	EXPECT_EQ('c', s.at(2));
+	EXPECT_EQ('d', s.at(3));
+
+	try
+	{
+		what = "";
+		s.at(4);
+	}
+	catch (std::out_of_range const& exc)
+	{
+		what = exc.what();
+	}
+	EXPECT_EQ(std::string("invalid string position"), what);
+
+	s.at(0) = 'd';
+	s.at(2) = 'b';
+	EXPECT_EQ("dbbd", s);
+}
+
+TEST(StringTest, TestElementAccessOperator)
+{
+	String s = "abcd";
+	EXPECT_EQ('a', s.operator[](0));
+	EXPECT_EQ('b', s.operator[](1));
+	EXPECT_EQ('c', s.operator[](2));
+	EXPECT_EQ('d', s.operator[](3));
+
+	s.operator[](0) = 'd';
+	s.operator[](2) = 'b';
+	EXPECT_EQ("dbbd", s);
+}
+
+TEST(StringTest, TestEmpty)
+{
+	String s = "abcd";
+	EXPECT_EQ(false, s.empty());
+
+	s = "";
+	EXPECT_EQ(true, s.empty());
+}
+
+TEST(StringTest, TestCStr) {
+	String s = "abcd";
+	s.c_str();
+	EXPECT_EQ("abcd", s);
+
+	s = "";
+	s.c_str();
+	EXPECT_EQ("", s);
+}
+
+TEST(StringTest, TestLength)
+{
+	String s = "abcd";
+	EXPECT_EQ(4, s.length());
+
+	s = "";
+	EXPECT_EQ(0, s.length());
+}
+
+TEST(StringTest, TestSize)
+{
+	String s = "abcd";
+	EXPECT_EQ(4, s.size());
+
+	s = "";
+	EXPECT_EQ(0, s.size());
+}
+
+TEST(StringTest, TestCapacity)
+{
+	String s = "abcd";
```

Comment:
I would add a check of size() for the string created by a default constructor. Don't remove the empty string case.